src/lib/storage/storage_manager.cpp

```diff
@@ -14,17 +14,17 @@ StorageManager& StorageManager::get() {
   return instance;
 }
 
-void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) { _tables.insert({name, table}); }
+void StorageManager::add_table(const std::string& name, std::shared_ptr<Table> table) {
+  DebugAssert(!has_table(name), "There already exists a table with the given name.");
```

Comment:
Same thing as above, I wont mark all the DebugAsserts now, just a little note that you might want to check this in the release build aswell.